pkgs/top-level/python-packages.nix

```
@@ -937,6 +937,8 @@ in {
 
   osmnx = callPackage ../development/python-modules/osmnx { };
 
+  OSMPythonTools = callPackage ../development/python-modules/OSMPythonTools { };
```

we try to follow pep8, which dictates that the module names should be lowercase
```suggestion
  osmpythontools = callPackage ../development/python-modules/osmpythontools { };
```

@FRidh is this still correct?

---

I'm a bit confused by you mentioning "module names" and PEP 8. My understanding is the following. (Please excuse that I'm stating some probably already obvious stuff while trying to sort this out.)

These names are involved here (and with upstream and this PR as-is all set to `OSMPythonTools`):
- the Python module name (what is used in `import` statements)
- the PyPI package name (what is used by `fetchPypi` or would be used when installing with `pip` or compatible tools from the Python ecosystem)
- the NixPkgs derivation name less the version (`pname`)
- the NixPkgs attribute name within `python3Packages`

As PEP 8 is for Python code and the Python ecosystem, its [naming conventions](https://pep8.org/#naming-conventions) per-se apply only to the first two and [would indeed mandate a lowercase name](https://pep8.org/#package-and-module-names). I don't think we should change the first two in NixPkgs. (In the case of the python module, that would cause incompatibilities with Python scripts importing that module using its upstream name. The PyPI package name doesn't seem to be case sensitive, though.) So if those were to change, that'd be the task of upstream and would mean a breaking change there.

For the latter two, while I see no reason why PEP 8 should apply to them (those are are a Nix identifier and a Nix value, respectively, not Python names), I'm not opposed to use all-lowercase names. I've initially chosen to use the same name as the PyPI package and Python module, because
- that seems to be what the library is known as
- containing an abbreviation, the name is hard to make sense of without any indication where words begin and end

Interestingly, the [GitHub repo's](https://github.com/mocnik-science/osm-python-tools) name is `osm-python-tools`.

Please advise whether:
- NixPkgs derivation name less the version (`pname`) should be changed
- NixPkgs attribute name within `python3Packages` should be changed

... and whether to `osmpythontools` or to some other value.

---

Attribute names in `python-packages.nix` *should* be the normalized names. Same goes for the folder. `pname` preferably as well. Note a different `pname` can be used for `fetchPypi`.
https://www.python.org/dev/peps/pep-0503/#normalized-names

---

so like @jonringer wrote

---

personally I like the normalized lowercase names, as I don't have to worry about capitalization when writing expressions.

---

Renamed attribute and derivation `pname` to lowercase with https://github.com/NixOS/nixpkgs/compare/204efd6757524f031e7e54be50907cf0d447b8c2..b73b61cbc881f9edf9f12ad0f2bec09b3da66c16

Kept `fetchPypi` `pname` mixed case.